membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -35,34 +37,38 @@ class AccountController extends LazyLogging {
     val tp = request.touchpoint
 
     (for {
-      user <- OptionT(Future.successful(authenticationService.userId))
-      sfUser <- OptionT(tp.contactRepo.get(user))
-      subscription <- OptionT(tp.subService.current[P](sfUser).map(_.headOption))
-      stripeCardToken <- OptionT(Future.successful(updateForm.bindFromRequest().value))
-      updateResult <- OptionT(tp.paymentService.setPaymentCardWithStripeToken(subscription.accountId, stripeCardToken))
+      user <- EitherT(Future.successful(authenticationService.userId \/> "no identity cookie for user"))
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_ \/> s"couldn't read contact from SF for $user (TODO check the separate ERROR to find out why)"))
```

Comment:
in membership common, we actually log some errors and then return a None.  It would make more sense to let the caller (i.e. us) deal with the error and log everything in one message.  I'd [already put "todos" there](https://github.com/guardian/membership-common/blob/8d579378aa1449040338553fd84a52cbe2825520/src/main/scala/com/gu/salesforce/ContactRepository.scala#L52-L58), as I had realised that was not right.